packages/apps/tools/TESTPLAN.md

```diff
@@ -0,0 +1,83 @@
+- **Preconditions:** User has a public key already created.
+- **Inputs:** User enters a valid key and adds to the list on clicking '+' next to the input.
+- **Expected Outcome:** The user enters a valid public key and the account name input is automatically filled with a k:account. The user clicks on the button and after a successful creation and funding, sees a notification that the transfer was successful to the given account.
+
+#### Scenario 2: Get error on entering invalid public key (length validation)
+- **Objective:** User cannot create an account with an invalid public key and fund 100 coins to it.
+- **Preconditions:** /
+- **Inputs:** User enters an invalid key and adds to the list on clicking '+' next to the input.
+- **Expected Outcome:** The user enters an invalid public key and the input shows error status. If the user clicks on the submit button sees a error text to provide valid public key(s) and cannot fund the given account.
+
+#### Scenario 3: Create multi key w:account and successfully fund 100 coins
+- **Objective:** User can create an account with multiple valid public keys and fund 100 coins to it.
+- **Preconditions:** User has two or more public keys already created.
+- **Inputs:** User enters valid keys and adds them to the list on clicking '+' next to the input.
+- **Expected Outcome:** The user enters multiple valid public keys and the account name input is automatically filled with a w:account. The user clicks on the button and after a successful creation and funding, sees a notification that the transfer was successful to the given account.
+
+
+
+### 3.3 [Faucet / Fund existing accounts]
+
+#### Scenario 1: Successfully fund 100 coins to an existing account
+- **Objective:** User can fund 100 coins to an existing account.
+- **Preconditions:** User has an account already created (on the selected chain).
+- **Inputs:** User enters a valid account name.
+- **Expected Outcome:** The user enters a valid account name. The user clicks on the submit button and after a successful funding, sees a notification that the transfer was successful to the given account.
+
+#### Scenario 2: Get error on entering invalid account name (length validation)
+- **Objective:** User cannot fund 100 coins to an invalid account name.
+- **Preconditions:** User has an account already created (on the selected chain).
+- **Inputs:** User enters an invalid account name.
+- **Expected Outcome:** The user enters an invalid account name (less than 3 or more than 256 characters) and the input shows error status. The submit button is disabled at this point.
+
+
+
+## 4. Test Data
+
+3.2 [Faucet / Create and fund new accounts]
+ - Scenario 1: any valid public key
+ - Scenario 2: any invalid public key
+ - Scenario 3: two or more valid public keys
+
+3.3 [Faucet / Fund existing accounts]
+- Scenario 1: valid account name created on the specific chain
+- Scenario 2: any invalid account name
+
+## 5. Dependencies
+
```

Comment:
Preferably usage of devnet should mentioned here, particularly for any form of integrated automated tests. Manually we can do the occasional check on testnet, but I'd hate to drain Sender00's account :D